clang/tools/clang-offload-bundler/ClangOffloadBundler.cpp

```diff
@@ -1434,7 +1447,7 @@ static Error UnbundleFiles() {
 
       // If this entry has a host kind, copy the input file to the output file
       // except for the archive unbundling where output is a list file.
-      if (hasHostKind(E.first()) && FilesType != "ao" && FilesType != "aoo")
+      if (hasHostKind(E.first()) && !FilesTypeIsArchive())
```

Comment:
I think for type "a" the current behavior should be preserved.

Reply:
good catch. Done